contracts/BondingMinter.sol

```diff
@@ -0,0 +1,33 @@
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import {IBondingMinter} from "./interfaces/IBondingMinter.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+contract BondingMinter is IBondingMinter {
+
+    constructor(uint _bondLimit,IERC721 _bondStorage) {
+        bondLimit = _bondLimit;
+        bondStorage = _bondStorage;
+    }
+
+    /* ========== CONSTANTS ========== */
+    uint immutable bondLimit;
+    IERC721 immutable bondStorage;
```

Comment:
We probably would need extended interface, not pure ERC721 since we'd need to get the position data from the token, so we need something like:

BondNFT immutable bondStorage;

Reply:
It means we need to implement BondNFT interface first